arch/arm/src/armv8-m/arm_stackcheck.c

```diff
@@ -69,6 +69,12 @@ void __stack_overflow_trap(void)
 
   uint32_t regval;
 
+  regval = getreg32(NVIC_DHCSR);
+  if (regval & 0x01)
```

Comment:
why need check NVIC_DHCSR

Reply:
Bit 0 in `NVIC_DHCSR` indicates whether debugging is active.

The `bkpt` instruction needs to be used only when debugging is enabled, or else a new fault is generated.

Reply:
it's better to add macro to nvic.h instead using magic number here directly.

Reply:
This is correct, I will change it.

Reply:
@xiaoxiang781216 See #7448 first, and then I will rebase.

Reply:
@xiaoxiang781216 changed it to use the register definition.